### PR TITLE
Add arrow-up support when choosing labels in a PR

### LINKED ISSUE #1450: Support pressing arrow-up when choosing labels
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/refined-github/issues/1450)

It's new feature proposal.

Scenario:
I created pull request, and I'm adding labels. I want to add label `Please CR: JS`. I type first two letters (`Pl`) into search input, and now I see 5 similar labels (e.g. `Please CR: CSS`, `Please CR: HTML`).  The one that I'm looking for is at the very end. I'd like to navigate to this label by pressing `up` arrow. Unfortunately, nothing happens.

Feature proposal:
During label searching, pressing `up` arrow at first item navigates me to the last one, and pressing `down` arrow at last item navigates me to the first item.

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/refined-github/issues/1450).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by idasbiste**
This feature adds support for cycled list navigation with keyboard support when choosing labels for PRs. 
The PR closes #1450 (I saw the project had [an older but closed PR](https://github.com/sindresorhus/refined-github/pull/1569) so I thought I could give it another try).

### CODE REVIEW

**Comment by sindresorhus** [comment]
Why limit it to labels? Could be applied to any list popover, like Reviewers, Assignees, Project, Milestones, etc.

**Comment by idasbiste** [comment]
@sindresorhus I limited it to labels because of the issue but I'm more than happy to expand it to the other lists! 👍

**Comment by Abraham6754** [comment]
Oh, sorry, hahaha, i think what the emails if for some projects un Github

File: source/content.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ import './features/filter-pr-by-build-status';
 import './features/edit-files-faster';
 import './features/update-pr-from-base-branch';
 import './features/hide-disabled-milestone-sorter';
+import './features/circular-movement-on-sidebar-lists';
```
Comment (bfred-it) [inline source/content.ts]:
```suggestion
import './features/cycle-lists-with-keyboard-shortcuts';
```

Comment (bfred-it) [inline source/content.ts]:
```suggestion
import './features/cycle-lists-with-keyboard-shortcuts';
```

File: source/features/circular-movement-on-sidebar-lists.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/**
+ * With this feature RGH will support cycled lists with keyboard support for arrow-up to navigate to the last item
+ * when pressed at the first one and for arrow-down to navigate to the first item when pressed at the last one
+ */
```
Comment (bfred-it) [inline source/features/circular-movement-on-sidebar-lists.tsx]: This description can be dropped

File: source/features/circular-movement-on-sidebar-lists.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function performSwapLabelBehaviour(event: KeyboardEvent, labels: HTMLElement[], from: number, to: number): void {
+	event.stopImmediatePropagation();
+	event.preventDefault();
+	swapLabelFocus(labels, from, to);
+}
+
+function init(): void {
+	document.addEventListener('keydown', event => {
```
Comment (bfred-it) [inline source/features/circular-movement-on-sidebar-lists.tsx]: This should use `delegate-it` like in my previous PR https://github.com/sindresorhus/refined-github/pull/1569/files (`onetime` is no longer needed though)

File: source/features/circular-movement-on-sidebar-lists.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+			select('.js-active-navigation-container .select-menu-list')!.scrollTop = 0;
+		}
+	});
+}
+
+features.add({
+	id: 'circular-movement-on-sidebar-lists',
+	description: 'Circular movement with arrow-up and -down keys on sidebar lists in compare view',
+	include: [
+		features.isCompare
```
Comment (bfred-it) [inline source/features/circular-movement-on-sidebar-lists.tsx]: This is the issue, it should work on a lot more pages

File: source/features/circular-movement-on-sidebar-lists.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		} else if (event.key === 'ArrowDown' && labels[lastLabelIndex].classList.contains('navigation-focus')) {
+			performSwapLabelBehaviour(event, labels, lastLabelIndex, 0);
+			select('.js-active-navigation-container .select-menu-list')!.scrollTop = 0;
+		}
+	});
+}
+
+features.add({
+	id: 'circular-movement-on-sidebar-lists',
+	description: 'Circular movement with arrow-up and -down keys on sidebar lists in compare view',
```
Comment (bfred-it) [inline source/features/circular-movement-on-sidebar-lists.tsx]:
```suggestion
	description: 'Cycle "select lists" (labels, etc) when selecting them with the `↑` and `↓` keys.',
```

**Comment by bfred-it** [comment]
Ping on this, if you have time

**Comment by bfred-it** [comment]
Feel free to reopen this if you have time to pick it up again

**Pull request closed without merging**